.github/workflows/pull_request_automation.yml

```diff
@@ -4,52 +4,36 @@ on:
   pull_request:
     branches: [ master, develop ]
 jobs:
-  linters:
+
+  tests:
     runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      max-parallel: 4
+      matrix:
+        python-version: ['3.6', '3.7', '3.8', '3.9']
 
     steps:
-
     - uses: actions/checkout@v2
 
-    - name: Set up Python
+    - name: Set up Python ${{ matrix.python-version }}
       uses: actions/setup-python@v2
       with:
-        python-version: 3.8
+        python-version: ${{ matrix.python-version }}
+
     - name: Install Dependencies
       run: |
-        sudo apt-get update
-        pip3 install --upgrade pip
-        pip3 install -e .[test]
-    - name: Black formatter
+        python3 -m pip install --upgrade pip
+        pip3 install tox tox-gh-actions flake8 codecov black
```

Comment:
These packages should be added to the `test-requirements.txt` file and then here we should only call `pip install -e .[test]`.